Feat/user event press (#1386)
callstack/react-native-testing-library: website/docs/UserEvent.md

### Options
- `delay` - controls the default delay between subsequent events, e.g. keystrokes, etc.
- `advanceTimers` - time advancement utility function that should be used for fake timers. The default setup handles both real and Jest fake timers.

## `press()`

```ts

> type(

Reviewer: ```suggestion
press(
```

  element: ReactTestInstance,
): Promise<void>
```

Example
```ts
const user = userEvent.setup();

> await user.press(touchable);

Reviewer: [nit] `element` name would seem more adequante here
```suggestion
await user.press(element);
```

```

> This helper simulates a press on any pressable element, e.g. Pressable, Text or TextInput. Unlike fireEvent.press which is a simpler API that will only call the onPress prop, this simumlates the entire press event in a more realistic way by reproducing what really happens when a user presses a pressable component. This means for instance for onPressIn and onPressOut props will also be invoked.  

Reviewer: ```suggestion
This helper simulates a press on any pressable element, e.g. `Pressable`, `TouchableOpacity`, `Text`, `TextInput`, etc. Unlike `fireEvent.press()` which is a simpler API that will only call the `onPress` prop, this simulates the entire press event in a more realistic way by reproducing what really happens when a user presses an interface view. This will trigger additional events like `pressIn` and `pressOut`. 
```

## `longPress()`

```ts

> type(

Reviewer: ```suggestion
longPress(
```

  element: ReactTestInstance,

>   options: { pressDuration: number } = { pressDuration: 500 }

Reviewer: [nit] Perhaps simple `duration` would be enough, as "press" should be evident from the context.

```suggestion
  options: { duration: number } = { duration: 500 }
```

Author: Yes i agree, I changed the option name

): Promise<void>
```

Example
```ts
const user = userEvent.setup();

> await user.longPress(touchable);

Reviewer: ditto: 
```suggestion
await user.longPress(element);
```
